### ISSUE TITLE
Blackout?

### ISSUE THREAD

**Issue by randomrileywat** [jwetzell/q-sys-plugin-novastar#4]
I just came across your plugin and wondered why blackout and normal working modes aren't included in the plugin. Would be super useful for us folks who want to blackout without affecting the brightness setting. Awesome plugin though.

**Comment by jwetzell** [comment]
I don't know what the control packets are for those commands on the various Novastar units. If you can find documentation or capture them from the network I can implement them.

**Comment by randomrileywat** [comment]
![Screenshot 2024-05-07 133637](https://github.com/jwetzell/q-sys-plugin-novastar/assets/126296524/b4aeee96-0421-457e-8689-55d0e1d3ff6e)
The Bitfocus Companion module has those controls for VX4S, VX6S, and VX1000.

The VX1000 blackout and normal code packets seem to work on the VX16S as well which I am working with this week.

**Comment by jwetzell** [comment]
Perfect, I will look at adding those!

**Comment by randomrileywat** [comment]
I will see what I can find out about VX16S as well. I know it uses the normal brightness, VX6S Input and preset commands, and VX1000 display modes so far. I don't know the actual connection packets though.

**Comment by jwetzell** [comment]
I have added the Normal, Freeze and Black display modes in this [branch](https://github.com/jwetzell/q-sys-plugin-novastar/blob/feat/blackout-freeze-modes/content/novastar.qplug). If you have the ability to test that this works that would be great.

**Comment by randomrileywat** [comment]
Absolutely! I have regular access to VX16S, VX1000, MCTRL4K, and UHDJr. driven walls. I will test out the branch tomorrow.

**Comment by randomrileywat** [comment]
Ended up being pretty busy today. I will be back on that site in a week or so. Apologies for the delay!

**Comment by jwetzell** [comment]
No worries!

**Comment by jwetzell** [comment]
@randomrileywat were you ever able to test these changes?

**Comment by randomrileywat** [comment]
I got stuck on some other projects, but I will be back on that site next week, and can do some more extensive testing. Thanks for the reminder!

**Comment by randomrileywat** [comment]
TLDR: Here are my notes from testing via emulation mode on QSYS v9.10.2. Some of the devices need some love. I was unable to get any of the devices to show as OK on the status. May be an issue with the initializing portion itself. I can also carve out some time to work on a branch if you would like some help with rearranging the input hex and all.

**VX1000**
Stuck on initializing.
Brightness works great.
Input switching works through DVI. (SDI, OPT 1&2, Mosaic don't work)
Test Patterns don't work
Presets recall, but lock the front panel as well.

**VX16S**
Tested on UHDJr. Layout
Stuck on initializing
Brightness works
DP Button = HDMI 1
HDMI Button = SDI 1
SDI1 Button = SDI 2
SDI2 Button = DVI 1
No Test Patterns Only one section took them.
Presets work

Also tested on VX6S Layout
Stuck on initializing
Presets work
HDMI 1 Button = HDMI 1
HDMI 2 Button = SDI 1
SDI1 Button = SDI 2
SDI2 Button = DVI 1
DVI1 Button = DVI 2
DVI2 Button = DVI 4
USB Button = Makes DVI 1-4 Flash

**NovaPro UHD Jr.**
Stuck on initializing
Brightness works
Presets work
Inputs 1-4 work
Missing DVI 1-4 inputs.
No test patterns.

**Pro-HD**
Stuck on initializing
Brightness works
Test patterns work!
SDI Button = DP
DVI Button = VGA
HDMI Button = CVBS
DP Button = VGA
VGA Button = DP
CVBS Button = SDI
Do ProHD's actually have presets??
Seems to only receive a command after around 30x presses. (Build this into code?)

**VX4S**
Stuck on initializing
Works flawlessly.
Brightness works!
Inputs work!
Test patterns work!
Do VX4S have presets?

**MCTRL4K**
Stuck on initializing.
Brightness works.
Inputs work, but missing DVI2.
Test patterns kind of work. (several clicks and only hits some outputs.)
Are there presets on MCTRL4K device?

**Comment by jwetzell** [comment]
Yeah, if you want to take a stab at the re-labeling that section is pretty straight forward. I've never had a NovaStar unit in hand to test this plugin on so the fact that they all do something is shocking to me. If you would open issues for some of the other non label related things (status not changing, missing inputs, broken test patterns, etc.) that would be great so I can track them all separately. I appreciate the thorough testing!

**Comment by mbhs-scott** [comment]
Any chance the VX4S-N could be updated to allow control of display freeze/on/off?

normal display
55 aa 00 00 fe 00 00 00 00 00 01 00 50 00 20 02 01 00 00 c7 56

Display freeze
55 aa 00 00 fe 00 00 00 00 00 01 00 50 00 20 02 01 00 01 c8 56

Display black
55 aa 00 00 fe 00 00 00 00 00 01 000 50 00 20 02 01 00 02 c9 56

[VX4S-Input-Switching-protocol.pdf](https://github.com/user-attachments/files/17484751/VX4S-Input-Switching-protocol.pdf)

**Comment by jwetzell** [comment]
@mbhs-scott have you tried out the changes in the [branch I mentioned above](https://github.com/jwetzell/q-sys-plugin-novastar/blob/feat/blackout-freeze-modes/content/novastar.qplug)? It looks like the display normal/freeze/black line up with what you posted. If it checks out I will merge it in.

**Comment by mbhs-scott** [comment]
@jwetzell Sorry mate I didn't see that part of the thread! I updated the plugin and now have the black/freeze/normal buttons and they work great for the VX4S-N that we have.
I've noticed that when I trigger the inputs however, it looks like a press of the normal mode is being sent at the same time. I'm not super great with scripting so havn't gone though it with a comb to see if that's the case, but can that be removed so its just the change in input?
It seems to constantly reconnect as well, but i'm sure I have seen that when connecting with other applications, the connection seems to go stale and needs prodding constantly to stay alive

**Comment by jwetzell** [comment]
No worries! Thanks for the quick test! The normal command was being set along with the input change. That has been removed now.

I've never had a Novastar unit to test this plugin with but the connection timing out wouldn't surprise me. I might have to do some research and see if there is some sort of keep alive I can send to avoid the timeout.

**Comment by mbhs-scott** [comment]
@jwetzell Wow you are awesome! Thanks so much for a quick change. I've tested and that is only sending the input change now. Thankyou
I look to have some issues with my Logic NOT workflows, but I can see the plugin is working well. Thanks!

**Comment by jwetzell** [comment]
Thanks for testing this out for me @mbhs-scott
